integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/Hooks.java

```diff
@@ -36,20 +43,41 @@ public void before(Scenario scenario) {
-        if (BridgeResource.getBridgeList(token).getItems().stream()
-                .anyMatch(b -> b.getId().equals(context.getBridgeId()))) {
-            BridgeResponse bridge = BridgeResource.getBridgeDetails(context.getManagerToken(), bridgeId);
-            if (bridge.getStatus() == BridgeStatus.READY) {
-                if (ProcessorResource.getProcessorList(token, bridgeId).getSize() > 0) {
-                    ProcessorResource.getProcessorList(token, bridgeId).getItems().stream()
-                            .forEach(p -> ProcessorResource.deleteProcessor(token, bridgeId, p.getId()));
-                    Awaitility.await().atMost(Duration.ofMinutes(2)).pollInterval(Duration.ofSeconds(5))
-                            .until(() -> ProcessorResource.getProcessorList(token, bridgeId).getSize() == 0);
-                }
-            }
-            BridgeResource.deleteBridge(token, bridgeId);
-        }
+        // Remove all bridges/processors created
+        context.getAllBridges().values()
+                .stream()
+                .filter(bridgeContext -> !bridgeContext.isDeleted())
+                .forEach(bridgeContext -> {
+                    final String bridgeId = bridgeContext.getId();
+                    BridgeResponse bridge = BridgeResource.getBridgeDetails(context.getManagerToken(), bridgeId);
+                    if (bridge.getStatus() == BridgeStatus.READY) {
+                        ProcessorListResponse processorList = ProcessorResource.getProcessorList(
+                                context.getManagerToken(),
+                                bridgeId);
+                        if (processorList.getSize() > 0) {
+                            processorList.getItems().stream().forEach(
+                                    p -> ProcessorResource.deleteProcessor(context.getManagerToken(), bridgeId,
+                                            p.getId()));
+                            Awaitility.await()
+                                    .atMost(Duration.ofMinutes(2))
+                                    .pollInterval(Duration.ofSeconds(5))
+                                    .until(() -> ProcessorResource.getProcessorList(context.getManagerToken(), bridgeId)
+                                            .getSize() == 0);
+                        }
+                    }
+                    switch (bridge.getStatus()) {
+                        case ACCEPTED:
+                        case PROVISIONING:
+                        case READY:
+                        case FAILED:
+                            try {
+                                BridgeResource.deleteBridge(context.getManagerToken(), bridgeId);
+                            } catch (Exception e) {
+                                LOGGER.warn(e, () -> "Unable to delete bridge with id " + bridgeId);
+                            }
+                        default:
+                            break;
+                    }
```

Comment:
To me this switch seems as in-appropriately used. Is there a status when we do not want to `deleteBridge` to be called?

Reply:
for those statuses, bridge is already in "delete" action

- DEPROVISION("deprovision")
- DELETING("deleting")
- DELETED("deleted")

Reply:
@radtriste thank you, I think I understand better now. My concenr is the big amount of enums/constants statuses. To preventa regression in future with new status added, my idea is to have to have some method like `public boolean isBeingDeleted()` directly in the enum, what do you think?

Reply:
@radtriste well, exception is really expansive operation, so if there is a way to avoid it, I vote for avoiding it, if you think we can not refactor currently your solution, I vote rather for merging as it is, maybe I will refactor later once I know codebase better.

Reply:
let's keep it like that and refactor it later. Not sure the statuses will change much as I think those are standard to MC

Reply:
can you please approve the PR ? :D